test/Record.purs

```diff
@@ -54,48 +65,57 @@ instance showProps ::
         | otherwise = str <> ", "
 
 showRecord :: forall r.
-  HFoldlWithIndex ShowProps String { | r } String =>
+  HFoldlWithIndex ShowProps String { | r } =>
   { | r } ->
   String
 showRecord r =
   "{ " <> hfoldlWithIndex ShowProps "" r <> " }"
 
-testShow :: _
-testShow = showRecord
-  { foo: "foo"
-  , bar: 42
-  , baz: false
-  }
+testShow :: Effect Unit
+testShow =
+  assertEqual
+    { actual:
+        showRecord
+          { foo: "foo"
+          , bar: 42
+          , baz: false
+          }
+    , expected:
+        "{ bar: 42, baz: false, foo: \"foo\" }"
+    }
 
-data ShowPropsCase = ShowPropsCase
 
-instance showPropsCase_Unit ::
-  (Show a, IsSymbol sym) =>
-  FoldingWithIndex ShowPropsCase (SProxy sym) Unit a String
-  where
-  foldingWithIndex _ prop _ a =
-    reflectSymbol prop <> ": " <> show a
+data ShowPropsCase = ShowPropsCase
 
-instance showPropsCase_String ::
+instance showPropsCase ::
   (Show a, IsSymbol sym) =>
-  FoldingWithIndex ShowPropsCase (SProxy sym) String a String
+  FoldingWithIndex ShowPropsCase (SProxy sym) String a
   where
-  foldingWithIndex _ prop pre a =
-    pre <> ", " <> reflectSymbol prop <> ": " <> show a
+  foldingWithIndex _ prop acc a =
+    pre <> reflectSymbol prop <> ": " <> show a
+    where
+    pre | acc == "" = ""
+        | otherwise = acc <> ", "
```

Comment:
Replaced separate `showPropsCase_Unit` and `showPropsCase_String` instances with a single instance that matches tutorial.